pkg/cmd/server/origin/openshift_apiserver.go

```diff
@@ -185,8 +186,24 @@ func (c *OpenshiftAPIConfig) SkipComplete() completedConfig {
 	return completedConfig{c}
 }
 
+func (c completedConfig) newTemplateConfig() *templateapiserver.TemplateConfig {
+	return &templateapiserver.TemplateConfig{
+		GenericConfig:       c.GenericConfig,
+		AuthorizationClient: c.KubeClientInternal.Authorization(),
+		Codecs:              kapi.Codecs,
+		Registry:            kapi.Registry,
+		Scheme:              kapi.Scheme,
+	}
+}
```

Comment:
> this seems like a strange place to have template specific code/logic, what's the design here?

It already relied on it (see storage.go, same call chain).  The openshift apiserver will be built from each apigroup's server so that the particulars of each api group's storage isn't contained in a single dependency magnet.  Done well, I think that we'll be able to make an interface for the separate server's construction.

Sharing the backing storage with the legacy api endpoint makes for some amount of centralization, but we don't have to have "normal" usage touching  a giant shared file that makes OWNERS really hard to describe.